### PR TITLE
add gvisor

### PR DESCRIPTION
**Pull request by rluria14**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rluria14** [comment]
@oblute be a maintainer?

**Comment by rluria14** [comment]
@conda-forge/staged-recipes this recipe is ready for review

File: recipes/gvisor/yum_requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+libstdc++-devel
```
Comment (xhochy) [inline recipes/gvisor/yum_requirements.txt:1]: This shouldn't be needed, we provide all C++ headers with the compiler.

**Comment by rluria14** [comment]
@conda-forge-admin please restart ci

**Comment by rluria14** [comment]
@conda-forge/staged-recipes this recipe is ready for review

**Comment by rluria14** [comment]
@conda-forge/staged-recipes ready for review

File: recipes/gvisor/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# Turn work folder into GOPATH
+export GOPATH=$PWD/gopath
+export PATH=${GOPATH}/bin:$PATH
+
+echo "module runsc" > go.mod
+
+go get gvisor.dev/gvisor/runsc@go
```
[the remark this answers is not in the document]
Comment (rluria14) [reply recipes/gvisor/build.sh]: @sodre took another approach and was able to get this built with bazel

File: recipes/gvisor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{% set provider = "gvisor" %}
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:1]: The provider variable makes more sense for terraform modules. It is not necessary in this case.
```suggestion
{% set name = "gvisor" %}
```

File: recipes/gvisor/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{% set goname = "github.com/google/"+provider.lower() %}
+{% set version = "20200323.0" %}
+
+{% set name = goname.split('/')[-1] %}
+{% set pkg_src = ('src/'+goname).replace("/",os.sep) %}
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:6]: `gname` and `pkg_src` were not being used anywhere else.
```suggestion
{% set version = "20200323.0" %}
```

File: recipes/gvisor/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/google/{{ provider }}/archive/release-{{ version }}.tar.gz
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:13]:
```suggestion
  url: https://github.com/google/{{ name }}/archive/release-{{ version }}.tar.gz
```

File: recipes/gvisor/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  sha256: 4a4b0d42ef741cfdb831ebc05ef6f483217515f6d186abc96af6fea3d1312353
+  patches:
+    - syscalls_rmv_boottime.patch
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:16]: It would be nice to have a comment why this patch is needed.

File: recipes/gvisor/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    - git
+    - bazel
+
+test:
+  commands:
+    - test //pkg/...
+    - test //runsc/...
+  source_files:
+    - pkg/
+    - runsc/
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:38]: The test section is a little bit problematic here.
  - What does `test //pkg/...` do? Did we mean `go test` ?
  - If we bring the entire `pkg` and `runsc` directories as source_files then the end go-package will be really big. Can we move that type of test to build.sh? 
  - Add a functional test, like `runsc --help` or equivalent, to the meta.yaml

File: recipes/gvisor/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,51 @@
+{% set provider = "gvisor" %}
+{% set goname = "github.com/google/"+provider.lower() %}
+{% set version = "20200323.0" %}
+
+{% set name = goname.split('/')[-1] %}
+{% set pkg_src = ('src/'+goname).replace("/",os.sep) %}
+
+package:
+  name: {{ name|lower }}
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:9]: Keep the go projects with go in their name.
```suggestion
  name: go-{{ name|lower }}
```

Comment (rluria14) [reply recipes/gvisor/meta.yaml]: @sodre all of these updates have been applied in the latest commit

File: recipes/gvisor/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - {{ compiler('go') }}
+    - git
+    - bazel
+  host:
+    - pip
+    - python
+  run:
+    - python
```
Comment (sodre) [inline recipes/gvisor/meta.yaml:31]: is python a requirement? 
```suggestion
```

Comment (rluria14) [reply recipes/gvisor/meta.yaml:31]: yes it's listed as a requirement on https://github.com/google/gvisor under installing from source

**Comment by rluria14** [comment]
@oblute do you agree to be a maintainer?

**Comment by sodre** [comment]
> @oblute do you agree to be a maintainer?

She had agreed in https://github.com/conda-forge/staged-recipes/pull/11183#issuecomment-607293949. 

I'm merging it.

**Comment by rluria14** [comment]
> > @oblute do you agree to be a maintainer?
> 
> She had agreed in [#11183 (comment)](https://github.com/conda-forge/staged-recipes/pull/11183#issuecomment-607293949).
> 
> I'm merging it.

thanks!

**Comment by jakirkham** [comment]
Thanks @rluria14! 😄

**Pull request merged**